Spigot-Server-Patches/0673-EntityExtinguishEvent.patch

```diff
@@ -0,0 +1,166 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Ivan Pekov <ivan@mrivanplays.com>
+Date: Mon, 1 Feb 2021 11:54:46 +0200
+Subject: [PATCH] EntityExtinguishEvent
+
+
+diff --git a/src/main/java/net/minecraft/server/BlockCauldron.java b/src/main/java/net/minecraft/server/BlockCauldron.java
+index 9fed3883828e7d6ca917a5eca7a7a3e37582f983..79d370abee1944defedb475584e1feeeb3bca037 100644
+--- a/src/main/java/net/minecraft/server/BlockCauldron.java
++++ b/src/main/java/net/minecraft/server/BlockCauldron.java
+@@ -1,6 +1,7 @@
+ package net.minecraft.server;
+ 
+ import org.bukkit.event.block.CauldronLevelChangeEvent; // CraftBukkit
++import io.papermc.paper.event.entity.EntityExtinguishEvent; // Paper
+ 
+ public class BlockCauldron extends Block {
+ 
+@@ -33,7 +34,7 @@ public class BlockCauldron extends Block {
+             if (!this.changeLevel(world, blockposition, iblockdata, i - 1, entity, CauldronLevelChangeEvent.ChangeReason.EXTINGUISH)) {
+                 return;
+             }
+-            entity.extinguish();
++            entity.extinguish(EntityExtinguishEvent.Reason.COLLISION); // Paper
+             // this.a(world, blockposition, iblockdata, i - 1);
+             // CraftBukkit end
+         }
+diff --git a/src/main/java/net/minecraft/server/Entity.java b/src/main/java/net/minecraft/server/Entity.java
+index f307a6361144c7e315b2e0ea45df27527cdb26ca..b799de0677054dfac8cc6a0a68bfd9a534131db3 100644
+--- a/src/main/java/net/minecraft/server/Entity.java
++++ b/src/main/java/net/minecraft/server/Entity.java
+@@ -5,6 +5,7 @@ import com.google.common.collect.Lists;
+ import com.google.common.collect.Sets;
+ import it.unimi.dsi.fastutil.objects.Object2DoubleArrayMap;
+ import it.unimi.dsi.fastutil.objects.Object2DoubleMap;
++import io.papermc.paper.event.entity.EntityExtinguishEvent; // Paper
+ import java.util.Arrays;
+ import java.util.Collection;
+ import java.util.Collections;
+@@ -557,7 +558,22 @@ public abstract class Entity implements INamableTileEntity, ICommandListener, Ke
+     }
+ 
+     public void extinguish() {
++        // Paper start - call EntityExtinguishEvent
++        extinguish(EntityExtinguishEvent.Reason.UNKNOWN);
```

Comment:
Prefer something like PLUGIN here

Reply:
I changed it to PLUGIN but further looking at it , extinguish() doesn't have it's own method in the API so it doesn't really make sense to be PLUGIN